timemachine/potentials/potential.py

```diff
@@ -77,3 +77,39 @@ def get_custom_ops_class_name_suffix(precision: Precision):
         return "f64"
     else:
         raise ValueError("invalid precision")
+
+
+def get_bound_potential_by_type(bps: Sequence[BoundPotential[_P]], pot_type: type[_P]) -> BoundPotential[_P]:
+    """Given a list of bound potentials return the first bound potential with the matching potential type.
+
+    Raises
+    ------
+        ValueError:
+            Unable to find potential with the expected type
+    """
+    result: Optional[BoundPotential[_P]] = None
+    for bp in bps:
+        if isinstance(bp.potential, pot_type):
+            result = bp
+            break
```

Comment:
nit: I think this is equivalent to
```suggestion
    result = next((bp for bp in bps if isinstance(bp.potential, pot_type)), None)
```
in case this seems cleaner to you. (I also think it's fine as is.)

Reply:
Find the version you suggested harder to read, as that None is not obvious as a default value. But I am far less experienced with the functional parts of python

Reply:
Seems worth adding a basic test asserting the expected behavior in the simple case as well as when the requested potential type is absent or appears multiple times? (Might have missed this, but would have expected to find it in `tests/test_potential.py` if it did exist)

Reply:
Added tests in https://github.com/proteneer/timemachine/pull/1374/commits/afaa0bdb9480045cf3682adb4a5103a8b51f213b